CHANGELOG.md

```diff
@@ -1,5 +1,18 @@
 # New Relic Ruby Agent Release Notes #
 
+  ## v8.7.0
+
+  * **Improved async support and Thread instrumentation**
+
+    Previously, the agent was not able to record events and metrics inside Threads created inside of an already running transaction. This release includes 2 new configuration options to support multithreaded applications to automatically instrument threads. To support applications that only want some threads instrumented by New Relic, a new class is available `NewRelic::TracedThread` that will create a thread that includes New Relic instrumentation, see our [API documentation](https://rubydoc.info/github/newrelic/newrelic-ruby-agent/NewRelic) for more details. 
```

Comment:
In addition to the rubydoc.info link, what do you think about adding `NewRelic::TracedThread` to our New Relic docs Ruby Custom Instrumentation docs too? https://docs.newrelic.com/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation

Reply:
Oh interesting, I hadn't considered that. Most of our API methods aren't described there, it seems like that is primarily describing how to do add_method_tracer and add_transaction_tracer stuff, which i think is the most common custom instrumentation usage. I'm not sure if this is something that should be added there or just in the API documentation. what do you think?

Reply:
Hmm... maybe this is a better question for the docs hero? 

@bradleycamacho - Should all of the Ruby Agent API methods be documented in two places -- once in the more Ruby-standard location and one on the New Relic site? Does this feature seem like it would fit within the scope of [this article](https://docs.newrelic.com/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation)?

Reply:
I'm open to either option, but my preferred method is keeping the info in the external ruby docs as long as:

1. `NewRelic::TracedThread` is more of an advanced configuration than a basic custom instrumentation
2. We add a more prominent link to the external ruby docs in our NR doc. Possibly in the intro or in a callout right below it.

The other option is creating a small section right under our [advanced configuration section](https://docs.newrelic.com/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation/#advanced-tracing). If we do this, we should ensure that it follows that pattern of being more of a full example/procedure instead of just defining it. What I'd want to avoid is just replicating content that may get out of sync (someone updates our doc and not the ruby docs or vice-versa).

Reply:
@tannalynn @fallwith - As a developer, it's more intuitive to me to update the RubyDoc site through `@api public` comments than adding something to the newrelic docs site. I'm leaning more toward #2, adding a more prominent link to the external ruby docs site in our NR doc. 

What are your thoughts?

I think I'm caring extra here because I want people to use this and know we have it!

Reply:
I'd say this is definitely more of an advanced custom instrumentation, I don't expect it would be one of the first things people would be looking for. So I think I also like the suggestion of adding the link to the rubydocs in a place that is more prominent. Also, since rubydocs are automatically updated when we release, it should always be fully up to date which is nice.

Reply:
I'm nodding along with all that's been brought up so far. Personally I'd find it useful to have a dedicated new section titled "Multithreading" / "Concurrency" / "Asynchronous Transactions" / "Threads" / etc. that clearly lay out the fact that we have 2 options: all threads, or just those we're explicitly told about, with source/configuration code block examples for each. It being a part of the advanced configuration section seems good.

Reply:
Excellent point! Perhaps that could be the new purpose for https://github.com/newrelic/newrelic-ruby-agent/issues/1057?

Reply:
Certainly!

Reply:
#1057 updated with this scope.